pxtwinrt/hiddeploy.ts

```diff
@@ -49,44 +53,78 @@ namespace pxt.winrt {
 
         initAsync(): Promise<void> {
             Util.assert(!this.dev, "HID interface not properly reseted");
-            const selector = Windows.Devices.HumanInterfaceDevice.HidDevice.getDeviceSelector(0xFF97, 0x0001);
-            return pxt.winrt.promisify(Windows.Devices.Enumeration.DeviceInformation.findAllAsync(selector, null)
-                .then(devices => {
-                    pxt.debug(`hid enumerate ${devices.length} devices`)
+            const wd = Windows.Devices;
+            const whid = wd.HumanInterfaceDevice.HidDevice;
+            if (!WindowsRuntimeIO.uf2Selectors) {
+                this.initUf2Selectors();
+            }
+            const getDevicesPromise = WindowsRuntimeIO.uf2Selectors.reduce((soFar, currentSelector) => {
+                // Try all selectors, in order, until some devices are found
+                return soFar.then((devices) => {
```

Comment:
New code works just like before: Get a list of all devices matching criteria, and connect to the 1st one. The difference is that before, we were not using VID / PID in the selectors, so we had a higher chance of attempting to connect to a wrong device (which would fail and say "Device not found").

Now that we use VID / PID to enumerate devices, only expected devices should be returned, and we will connect to the first one